pkg/prometheus/promcfg.go

```diff
@@ -3275,6 +3275,86 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// DockerSDConfig
+	if len(sc.Spec.DockerSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.DockerSDConfigs))
+
+		for i, config := range sc.Spec.DockerSDConfigs {
+			assetStoreKey := fmt.Sprintf("scrapeconfig/%s/%s/dockersdconfig/%d", sc.GetNamespace(), sc.GetName(), i)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/dockersdconfig/%d", sc.GetNamespace(), sc.GetName(), i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], config.OAuth2, store.OAuth2Assets, assetStoreKey)
+			configs[i] = cg.addProxyConfigtoYaml(ctx, configs[i], sc.GetNamespace(), store, config.ProxyConfig)
+			configs[i] = cg.addBasicAuthToYaml(configs[i], fmt.Sprintf("scrapeconfig/%s/%s/dockersdconfig/%d", sc.Namespace, sc.Name, i), store, config.BasicAuth)
+
+			if config.TLSConfig != nil {
+				configs[i] = addSafeTLStoYaml(configs[i], sc.GetNamespace(), *config.TLSConfig)
+			}
+            // TODO: Should we default to a value?
+			if config.Port != nil {
+				configs[i] = append(configs[i], yaml.MapItem{
+					Key:   "port",
+					Value: config.Port,
+				})
+			}
+
+            // TODO: Should we default to a value?
```

Comment:
default will be handled by Prometheus, its better we don't provide default since it can changed in Prometheus. So its better Prometheus handle default value set by Prometheus

Reply:
Removed the comment, resolving.